scipy/integrate/tests/test_tanhsinh.py

```diff
@@ -752,8 +752,13 @@ def test_gh_22681_finite_error(self, xp):
         b = complex(12, 39)
         def f(t):
             return xp.sin(a * (1 - t) + b * t)
-        res = _tanhsinh(f, xp.asarray(0.), xp.asarray(1.), atol=0, rtol=0, maxlevel=10)
-        assert xp.isfinite(res.error)
+        ref = _tanhsinh(f, xp.asarray(0.), xp.asarray(1.), atol=0, rtol=0, maxlevel=10)
+        assert xp.isfinite(ref.error)
+        # Previously, the tanhsinh would not detect convergence
+        res = _tanhsinh(f, xp.asarray(0.), xp.asarray(1.))
```

Comment:
*This* wouldn't pass in main, but does pass in this PR.
```suggestion
        res = _tanhsinh(f, xp.asarray(0.), xp.asarray(1.), rtol=1e-14)
```